packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -19,11 +19,19 @@ export type ObjectFromSchemaRecord<
 };
 
 /**
- * Helper used to produce types for object nodes.
+ * A {@link TreeNode} which modules a JavaScript object.
+ * @remarks
+ * Object nodes consist of a type which specifies which {@link TreeNodeSchema} they use (see {@link TreeNodeApi.schema}), and a collections of fields, each with a distinct `key` and its own {@link FieldSchema} defining what can be placed under that key.
+ *
+ * All non-empty fields on an object node are exposed as enumerable own properties with string keys.
+ * No other own `own` or `enumerable` properties are included on object nodes unless the user of the node manually adds custom session only state.
+ * This allows a majority of general purpose JavaScript object processing operations (like `for...in`, `Reflect.ownKeys()` and `Object.entries()`) to enumerate all the children.
```

Comment:
@noencke I think this currently isn't true.

Reply:
Hopefully it is true or truer after the other PR goes in :)